docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Deploy using Helm.
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## KafkaApp
+
+Sub class of [_KubernetesApp_](#kubernetesapp).
+
+### Usage
+
+- Deploys a [streams-bootstrap](https://github.com/bakdata/streams-bootstrap)
+component
+- Should not be used in `pipeline.yaml` as the component can be defined as either a
+[StreamsApp](#streamsapp) or a [Producer](#producer)
+- Often used in `defaults.yaml`
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+In addition to [KubernetesApp's](#kubernetesapp) `deploy`:
+
+- Create topics if provided (optional)
+- Submit schemas to the registry if provided (optional)
+
+#### destroy
+
+Refer to [KubernetesApp](#kubernetesapp).
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## StreamsApp
+
+Sub class of [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Configures a
+[streams-bootstrap app](https://github.com/bakdata/streams-bootstrap) which in
+turn configures a
+[Kafka Streams app](https://kafka.apache.org/documentation/streams/).
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/streams-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### destroy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### reset
+
+Runs a reset job in the cluster that is deleted by default after a succesful
+completion.
+
+- Reset the consumer group offsets
+- Reset Kafka Streams state
+
+#### clean
+
```

Comment:
```suggestion
Runs a cleanup job in the cluster that is deleted by default after a successful
completion.
```

Reply:
I think this belongs to [cli-commands](https://github.com/bakdata/kpops/blob/main/docs/docs/user/references/cli-commands.md) as it is a general description of `clean`. It would also be mentioned in the yamls page as setting the behavior of `clean` is part of `config.yaml`

btw, since it is set with an env variable, shouldn't it be mentioned in [variables](https://github.com/bakdata/kpops/blob/main/docs/docs/user/references/variables.md)?

Reply:
I added this  here because you have the same sentence for the `reset`

Reply:
Maybe add an issue on updating the variables section.

Reply:
None of the env variables defined in the config schema are present on the page, are they supposed to be there?

Reply:
I suppose this config environment variables should go on a separate documentation page for `config.yaml`. The env variables which are covered so far belong to the CLI

Reply:
https://github.com/bakdata/kpops/issues/129